### PR TITLE
[next-auth] Add JWTOptions.verificationOptions

### PR DESCRIPTION
**Pull request by recurser**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nextauthjs/next-auth/blob/76df2b5e702be2084674c81e098d989c2ec39ed9/src/lib/jwt.js#L58-L61
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

### CODE REVIEW

File: types/next-auth/jwt.d.ts [hunk 1]
```diff
@@ -40,6 +40,10 @@ export interface JWTOptions {
     encryptionKey?: string;
     encode?(options: JWTEncodeParams): Promise<string>;
     decode?(options: JWTDecodeParams): Promise<WithAdditionalParams<JWT>>;
+    verificationOptions?: {
+      algorithms?: string[]
+      maxTokenAge?: string
```
Comment (recurser) [inline types/next-auth/jwt.d.ts:45]: @balazsorban44 do you think `algorithms` and  `maxTokenAge` should both be optional? Or both are required?

Comment (lluia) [reply types/next-auth/jwt.d.ts:45]: `verificationOptions` should be optional given [in the source code](https://github.com/nextauthjs/next-auth/blob/76df2b5e702be2084674c81e098d989c2ec39ed9/src/lib/jwt.js#L58-L61) we use a default parameter for it 📐

Regarding `algorithms` and `maxTokenAge`, [they're forwarded](https://github.com/nextauthjs/next-auth/blob/76df2b5e702be2084674c81e098d989c2ec39ed9/src/lib/jwt.js#L90) to [`jose`](https://www.npmjs.com/package/jose), looking at [its documentation ](https://github.com/panva/jose/blob/dd6b0eda9f07af21fb4d650fbc22ff5bd23bded4/docs/interfaces/jwt_verify.jwtverifyoptions.md) it seems both are optional 👍🏽

Comment (lluia) [reply types/next-auth/jwt.d.ts:45]: Just a nitpick that according to [`jose` docs](https://github.com/panva/jose/blob/dd6b0eda9f07af21fb4d650fbc22ff5bd23bded4/docs/interfaces/jwt_verify.jwtverifyoptions.md), `maxTokenAge` is actually `string | number` 🏂

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Just a small 💬 , thanks a lot 💚

File: types/next-auth/jwt.d.ts (hunk 1 above)
Comment (recurser) [reply types/next-auth/jwt.d.ts]: @lluia thanks! Updated 🙇

**Comment by recurser** [comment]
@lluia I don't know if you have any insights here, but I'm a bit mystified why the tests have suddenly started failing. I get the same failures locally on `master` when testing `next-auth`, so it doesn't seem to be specific to my changes. This PR was passing previously, and git-bisect is not giving any clues. Any ideas?

**Comment by lluia** [comment]
@recurser not sure either 🤔 , I'll investigate 👍🏽

**Comment by dperrett** [comment]
@lluia thanks! 🙇

**Comment by lluia** [comment]
@dperrett we were on an effort last week [to move these types to the `next-auth` package](https://github.com/nextauthjs/next-auth/issues/1649) for easier maintainability 🙌🏽

They are for now only available on the latest `canary` release of `next-auth`: `next-auth@3.14.0-canary.7` 📦 . We plan to finish the last migration tasks next week and deprecate the types here accordingly.

Would you mind [moving your changes there](https://github.com/nextauthjs/next-auth/tree/beta/types)? You can open a PR against the `beta` branch and tag @balazsorban44 and me 🙏🏽 for review. Thanks for the help and sorry for the inconvenience 🙇🏽‍♂️

**Comment by recurser** [comment]
@lluia sure thing! 👍 I'll close this one for now and continue in `next-auth` directly.

**Comment by recurser** [comment]
For reference, I've submitted a PR [here](https://github.com/nextauthjs/next-auth/pull/1715).

**Pull request closed without merging**